mlflow/models/dependencies_schema.py

```diff
@@ -58,13 +59,39 @@ def _get_vector_search_schema():
     Returns:
         VectorSearchIndex: The vector search index schema.
     """
-    return VectorSearchIndexSchema(
-        name="vector_search_index",
-        primary_key=globals().get(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY),
-        text_column=globals().get(DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN),
-        doc_uri=globals().get(DATABRICKS_VECTOR_SEARCH_DOC_URI),
-        other_columns=globals().get(DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS),
+    if not globals().get(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY) or not globals().get(
+        DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN
+    ):
+        return []
+
+    return [
+        VectorSearchIndexSchema(
+            name="vector_search_index",
+            primary_key=globals().get(DATABRICKS_VECTOR_SEARCH_PRIMARY_KEY),
+            text_column=globals().get(DATABRICKS_VECTOR_SEARCH_TEXT_COLUMN),
+            doc_uri=globals().get(DATABRICKS_VECTOR_SEARCH_DOC_URI),
+            other_columns=globals().get(DATABRICKS_VECTOR_SEARCH_OTHER_COLUMNS),
+        )
+    ]
+
+
+def _clear_dependencies_schema():
+    """
+    Clear all the dependencies schema defined by the user.
+    """
+    # Clear the vector search schema
+    set_vector_search_schema(primary_key=None, text_column=None, doc_uri=None, other_columns=None)
+
+
+@contextmanager
+def _get_dependencies_schema():
```

Comment:
This is a little bit weird if I set_vector_search_schema, then using this context manager will erase all schemas set. Would it be better if we have a something like:
```
@contextmanager
def set_dependencies_schema(vector_search_index_schema):
        yield DependenciesSchemas(vector_search_index_schemas=[vector_search_index_schema])
```
Not a blocker though if you need this kind of mechanism!

Reply:
Unfortunately we ask users to call `set_vector_search_schema` and add it to a global variable.
later on we read the global variable via the context here and write it to a file.

If there is a better way to do this, I am happy to do that in a follow up if that works? :)